src/psyclone/psyir/nodes/array_of_structures_reference.py

```diff
@@ -118,12 +120,16 @@ def create(symbol, indices=None, members=None, parent=None):
                 "An ArrayOfStructuresReference must refer to a symbol of "
                 "ArrayType but symbol '{0}' has type '{1}".format(
                     symbol.name, symbol.datatype))
-        if not isinstance(indices, list) or not indices:
+        if indices is None:
```

Comment:
Previously `indices` was mandatory but when I came to implement the frontend support in this PR I realised that I wanted to use `process_nodes()` to handle the array index expressions. In order for that to work, I needed to be able to create the parent `ArrayOfStructuresReference` first. Therefore, I had to alter `create()` so that `indices` is an optional argument. `members` is always mandatory. Does that make sense?

Reply:
For `members`  here by mandatory I was meaning removing the None in the `members=None` method signature. But I see this can't be done unless indices has the None removed first (as I still want to keep this ordering).

So then, about indices, the fparser2 can be done once again with a fakeparent ArrayReference node trick, right? I see this is not the nicest trick but then it's a question that if the create method should be nicer (always finish with a consistent/complete state) or the fparser2 frontend should have nicer code (avoid fakeparent tricks). One could argue that the fparser2 already uses this trick often but I don't know where the balance should be.  I will let you decide.

Reply:
Oh yes - I only thought of the fake parent trick after I'd implemented this. I've changed the code now and made `indices` and `members` mandatory.